src/workerd/api/r2-bucket.h

```diff
@@ -492,6 +522,9 @@ class R2Bucket: public jsg::Object {
         options?: R2PutOptions & { onlyIf: R2BucketConditional | Headers }
       ): Promise<R2Object | null>;
       put(key: string, value: ReadableStream | ArrayBuffer | ArrayBufferView | string | null | Blob, options?: R2PutOptions): Promise<R2Object>;
+
+      copy(key: string, source: R2CopySource, options: R2CopyOptions & { onlyIf: R2BucketConditional | Headers }): Promise<R2Object | null>;
```

Comment:
Where does R2BucketConditional come from?

Reply:
I'm not entirely sure how, but it is converted to `R2Conditional` when the types are built. I've asked the team to make sure I understand